src/journal/src/Journal/Internal/ByteBuffer.hs

```diff
@@ -105,11 +105,17 @@ remaining bb = do
 ------------------------------------------------------------------------
 -- * Checks
 
-boundCheck :: ByteBuffer -> Int -> IO ()
-boundCheck bb ix = do
-  if fromIntegral ix <= getCapacity bb
+boundCheck :: ByteBuffer -> String -> Int -> IO ()
```

Comment:
Use `HasCallStack` to get who called instead of the `ctx :: String`? Could maybe even get more of the callstack?